go-controller/pkg/ovn/controller/admin_network_policy/types.go

```diff
@@ -0,0 +1,392 @@
+)
+
+// TODO: Double check how empty selector means all labels match works
+type adminNetworkPolicySubject struct {
+	namespaceSelector labels.Selector
+	podSelector       labels.Selector
+	// map of namespaces matching the provided namespaceSelector
+	// {K: namespace name; V: {set of pods matching the provided podSelector}}
+	namespaces map[string]sets.Set[string]
+	// all the LSP port UUIDs of the subject pods selected by this ANP
+	// Since nbdb.PortGroup.LSP stores the UUIDs of the LSP's storing it
+	// in the cache makes it easier to compute the difference between
+	// current set of UUIDs and desired set of UUIDs and do one set of
+	// transact ops calculation. If not, for every pod/namespace update
+	// we would need to do a lookup in the libovsdb cache for the ns_name
+	// LSP index. TODO(tssurya): Do performance runs to see if there is
+	// effect on MEM footprint for storing this information.
+	podPorts sets.Set[string]
+}
+
+// TODO: Implement sameLabels & notSameLabels
+type adminNetworkPolicyPeer struct {
+	namespaceSelector labels.Selector
+	podSelector       labels.Selector
+	// map of namespaces matching the provided namespaceSelector
+	// {K: namespace name; V: {set of pods matching the provided podSelector}}
+	namespaces map[string]sets.Set[string]
+}
+
+type adminNetworkPolicyPort struct {
+	protocol string
+	port     int32 // will store startPort if its a range
+	endPort  int32
+}
+
+type gressRule struct {
+	name string
+	// priority is determined based on order in the list and calculated using adminNetworkPolicyState.priority
+	priority int32
+	// gressIndex tracks the index of this rule
+	gressIndex  int32
+	gressPrefix string
+	// NOTE: Action here is the corresponding OVN action for
+	// anpapi.AdminNetworkPolicyRuleAction
+	action string
+	peers  []*adminNetworkPolicyPeer
+	ports  []*adminNetworkPolicyPort
+	// all the podIPs of the peer pods selected by this ANP Rule
+	podIPs sets.Set[string]
+	// address-set that contains the IPs of all peers in this rule
+	addrSet addressset.AddressSet
+}
+
+// adminNetworkPolicyState is the cache that keeps the state of a single
+// admin network policy in the cluster with name being unique
+type adminNetworkPolicyState struct {
+	// name of the admin network policy (unique across cluster)
+	name string
+	// priority is the OVN priority equivalent of anp.Spec.Priority
+	priority int32
+	// subject stores the objects needed to track .Spec.Subject changes
+	subject *adminNetworkPolicySubject
+	// ingressRules stores the objects needed to track .Spec.Ingress changes
+	ingressRules []*gressRule
+	// egressRules stores the objects needed to track .Spec.Egress changes
+	egressRules []*gressRule
+}
+
+// cloneANP takes the provided ANP API object and creates a new corresponding
+// adminNetworkPolicyState cache object for that API object.
+func cloneANP(raw *anpapi.AdminNetworkPolicy) (*adminNetworkPolicyState, error) {
```

Comment:
Suggesting `newAdminNetworkPolicyState` instead of `cloneANP`. Similar for the others.

Reply:
ack changing.. (I know you had pointed it out in your first reviews but I couldn't come up with a better name :D ), I am ok with `new-XX` ; changing this

Reply:
this is done.